src/libraries/Microsoft.PowerFx.Core/Functions/Delegation/DelegationStrategies/DelegationValidationStrategy.cs

```diff
@@ -405,9 +407,16 @@ protected bool IsValidCallNodeInternal(CallNode node, TexlBinding binding, Opera
                 SuggestDelegationHint(node, binding, warning, new object[] { callInfo?.Function.Name });
             }
 
-            if (callInfo?.Function != null && ((TexlFunction)callInfo.Function).IsRowScopedServerDelegatable(node, binding, metadata))
+            if (callInfo?.Function != null && ((TexlFunction)callInfo.Function).IsRowScopedServerDelegatable(node, binding, metadata, nodeInheritsRowScope: nodeInheritsRowScope))
             {
                 return true;
+            }
+
+            if (callInfo?.Function is UserDefinedFunction udf &&
+                ((FilterFunction)trackingFunction).IsValidDelegatableFilterPredicateNode(udf.Binding.Top, udf.Binding, metadata as FilterOpMetadata, nodeInheritsRowScope: isRowScoped))

```

Comment:
>FilterFunction [](http://example.com/codeflow?start=18&length=14)

Can we guarantee that this is going to be a FilterFunction?

Reply:
No, sorry that was an outdated commit